pam_tcb/support.h

```diff
@@ -147,10 +147,19 @@ struct unix_verify_password_param {
 	const char *pass;
 };
 
+#define pam_tcb_overwrite_string(xx)					\
+do {									\
+	void *xx__ = xx;						\
+	if (xx__) {							\
+		xx__ = memset(xx__, '\0', strlen(xx__));		\
+		__asm__ __volatile__ ("" : : "r"(xx__) : "memory");	\
```

Comment:
I don't mind the asm trick in _this_ project, which is Linux-specific anyway. But we need to also come up with some generic way for other projects, such as `passwdqc`.

Reply:
Linux-PAM uses `memset_explicit` or `explicit_bzero` if they are available, and uses this asm trick as a fallback.
We could at least do the same if we had a way to autodetect available functions at build time.